functional-tests/tests/prover/prover_client_restart.py

```diff
@@ -0,0 +1,69 @@
+import time
+
+import flexitest
+
+from envs import testenv
+from envs.testenv import BasicEnvConfig
+from utils import (
+    ProverClientSettings,
+    RollupParamsSettings,
+    wait_for_proof_with_time_out,
+    wait_until,
+)
+
+
+@flexitest.register
+class ProverClientRestartTest(testenv.StrataTester):
+    def __init__(self, ctx: flexitest.InitContext):
+        # A separate standalone env for this test as it involves a restart
+        # and the rollup settings are non-standard.
+
+        # Increase the proof timeout so that the checkpoint index increments only
+        # after the prover client submits the corresponding checkpoint proof
+        rollup_settings = RollupParamsSettings.new_default()
+        rollup_settings.proof_timeout = 300
+
+        ctx.set_env(
+            BasicEnvConfig(
+                pre_generate_blocks=101,
+                prover_client_settings=ProverClientSettings.new_with_proving(),
+                rollup_settings=rollup_settings,
+            )
+        )
+
+    def main(self, ctx: flexitest.RunContext):
+        prover_client = ctx.get_service("prover_client")
+        prover_client_rpc = prover_client.create_rpc()
+        sequencer = ctx.get_service("sequencer")
+        sequencer_rpc = sequencer.create_rpc()
+
+        # Wait for the Prover Manager setup
+        time.sleep(5)
```

Comment:
Do we have some kind of RPC endpoint that indicates the client is running? We could use that instead of having a hard sleep here.

Reply:
But anyways, given the pretty big waiting times for creating proofs, this should already be negligible.

Reply:
Yeah I'm also in favor for that.

Reply:
done